libse/SubtitleFormats/FinalCutProXml15.cs

```diff
@@ -438,24 +444,24 @@ public override void LoadSubtitle(Subtitle subtitle, List<string> lines, string
                     {
                         try
                         {
-                            string text = node.ParentNode.InnerText.Replace("\r\r", "\r");
-                            var p = new Paragraph();
-                            p.Text = text.Trim();
+                            string text = node.ParentNode.InnerText.Replace("\r\r", "\r").Trim();
                             if (node.ParentNode.InnerXml.Contains("bold=\"1\""))
-                                p.Text = "<b>" + p.Text + "</b>";
+                                text = "<b>" + text + "</b>";
                             if (node.ParentNode.InnerXml.Contains("italic=\"1\""))
-                                p.Text = "<i>" + p.Text + "</i>";
+                                text = "<i>" + text + "</i>";
+                            var p = new Paragraph();
                             p.StartTime = DecodeTime(node.ParentNode.Attributes["offset"]);
                             p.EndTime.TotalMilliseconds = p.StartTime.TotalMilliseconds + DecodeTime(node.ParentNode.Attributes["duration"]).TotalMilliseconds;
-                            bool add = true;
+                            p.Text = text;
                             if (subtitle.Paragraphs.Count > 0)
                             {
                                 var prev = subtitle.Paragraphs[subtitle.Paragraphs.Count - 1];
-                                if (prev.Text == p.Text && prev.StartTime.TotalMilliseconds == p.StartTime.TotalMilliseconds)
-                                    add = false;
+                                if (prev.Text.Equals(p.Text, StringComparison.Ordinal) && prev.StartTime.TotalMilliseconds == p.StartTime.TotalMilliseconds)
+                                {
+                                    continue; // ignore duplicated paragraphs
```

Comment:
Why have you added this?

Reply:
We don't need variable **add** to check if paragraph should be added

```c#
var p = new Paragraph();
p.StartTime = DecodeTime(node.ParentNode.Attributes["offset"]);
p.EndTime.TotalMilliseconds = p.StartTime.TotalMilliseconds + DecodeTime(node.ParentNode.Attributes["duration"]).TotalMilliseconds;
p.Text = text;
if (subtitle.Paragraphs.Count > 0)
{
	var prev = subtitle.Paragraphs[subtitle.Paragraphs.Count - 1];
	if (prev.Text.Equals(p.Text, StringComparison.Ordinal) && prev.StartTime.TotalMilliseconds == p.StartTime.TotalMilliseconds)
	{
		continue; // ignore duplicated paragraphs
	}
}
subtitle.Paragraphs.Add(p);
```